pkg/syncthing/syncthing.go

```diff
@@ -329,64 +332,44 @@ func (s *Syncthing) WaitForScanning(ctx context.Context, wg *sync.WaitGroup, dev
 // WaitForCompletion waits for the remote to be totally synched
 func (s *Syncthing) WaitForCompletion(ctx context.Context, wg *sync.WaitGroup, dev *model.Dev, reporter chan float64) error {
 	defer close(reporter)
-	ticker := time.NewTicker(500 * time.Millisecond)
-	folder := fmt.Sprintf("okteto-%s", dev.Name)
-	params := map[string]string{"folder": folder, "device": DefaultRemoteDeviceID}
-	completion := &Completion{}
-	var prevNeedBytes int64
-	needZeroBytesIter := 0
-	log.Infof("waiting for synchronization to complete...")
-	for {
-		select {
-		case <-ticker.C:
-			if prevNeedBytes == completion.NeedBytes {
-				if needZeroBytesIter >= 50 {
-					return errors.ErrSyncFrozen
-				}
-
-				needZeroBytesIter++
-			} else {
-				needZeroBytesIter = 0
-			}
-		case <-ctx.Done():
-			log.Debug("cancelling call to 'rest/db/completion'")
-			return ctx.Err()
-		}
-
+	params := map[string]string{
+		"events":  "FolderCompletion",
+		"device":  DefaultRemoteDeviceID,
+		"folder":  fmt.Sprintf("okteto-%s", dev.Name),
+		"timeout": "0",
+		"since":   "0",
+	}
+	var since int
+	var completion float64
+	log.Info("waiting for synchronization to complete...", completion)
+	for completion != 100 {
```

Comment:
At some point we've seen the syncthing return a 100% completion and then go back to 0 and then start increasing (it was a problem that the companion app ran into). I believe we were using this same api. Have you seen anything like it?

Reply:
that was because we were calling completion twice, before and after overwriting local changes.